spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/data/neo4j/Neo4jDataAutoConfigurationTests.java

```diff
@@ -175,6 +197,21 @@ public SessionFactory customSessionFactory() {
 	}
 
 	@Configuration
+	@EnableConfigurationProperties(Neo4jProperties.class)
+	static class CustomConfigurationFactory {
+
+		@Bean
+		public org.neo4j.ogm.config.Configuration configuration(Neo4jProperties properties) {
```

Comment:
That's a lot of code to check that `Neo4jProperties` behave as expected. You could just as well create the object by hand, create the configuration and then check it has done the expected thing (plain unit test that is).